### ISSUE TITLE
Submission views with deleted quest causes errors

### ISSUE THREAD

**Issue by tylerecouture** [bytedeck/bytedeck#1488]
This error was introduced because we don't have any basic tests checking if views are accessible when a QuestSubmission's quest has been deleted (= None).  So we need those tests at lease for the Submission Detail View  and then fix error below:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/exception.py", line 47, in inner
    response = get_response(request)
  File "/usr/local/lib/python3.8/site-packages/django/core/handlers/base.py", line 181, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
  File "/app/src/tenant/views.py", line 40, in wrapper
    return f(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/contrib/auth/decorators.py", line 21, in _wrapped_view
    return view_func(request, *args, **kwargs)
  File "/app/src/quest_manager/views.py", line 1301, in submission
    if sub.quest.xp_can_be_entered_by_students and not sub.is_approved:

Exception Type: AttributeError at /quests/submission/103044/
Exception Value: 'NoneType' object has no attribute 'xp_can_be_entered_by_students'

**Comment by yujinyuz** [comment]
@tylerecouture I need your opinion on this one.

I currently added a `messages.warning` if the quest does not exist and redirect them to the quests page.

The problem now is that quests don't get displayed

<img width="903" alt="image" src="https://github.com/bytedeck/bytedeck/assets/10972027/986ba72e-57bf-4db8-b371-abfad231b803">

And the in progress tab looks like this

<img width="914" alt="image" src="https://github.com/bytedeck/bytedeck/assets/10972027/1c44a49f-cfe0-4ef9-9cc4-80838a700f31">

---

Should we automatically drop any quest submissions when a quest gets deleted? OR should we prevent a quest from being deleted if there are existing submissions?

**Comment by tylerecouture** [comment]
@yujinyuz I think that would make things easier.  However how do we handle all the currently orphaned QuestSubmissions where Quest=None?

Is this part of a migration where we delete them all before changing the on_delete behaviour of the foreign key?

**Comment by yujinyuz** [comment]
@tylerecouture Yup. 2 migration scripts will be created. 1 for pruning orphaned quests and 1 for the delete cascade on quest.

I took a look into the commit history as to why it the `quest` inside `QuestSubmission` was set to `on_delete=models.SET_NULL` but I couldn't get any info as the commit message only says `resolve django2 warnings`

https://github.com/bytedeck/bytedeck/commit/b3fde3e7b4c2c6f5a34008c4851744704b967a69

---

I have another concern. I noticed that there are some test cases that test if a quest is null for a quest submission. Would it be okay if we remove those test cases?
